include/extractor/guidance/turn_handler.hpp

```diff
@@ -45,9 +47,24 @@ class TurnHandler : public IntersectionHandler
                             Intersection intersection) const override final;
 
   private:
+    struct Fork
+    {
+        Intersection::iterator right;
+        Intersection::iterator left;
+        const std::size_t size;
+        Fork(Intersection::iterator right, Intersection::iterator left);
```

Comment:
Fork handling is only called on Intersections (in the dedicated handlers), just as the sliproad handler. Right now this is correct. We could think about how we could update this behaviour, but right no I'd say for the scope of this PR having Intersections here is correct.